azafea/event_processors/metrics/events/__init__.py

```diff
@@ -165,6 +165,79 @@ def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
         return {'app_id': payload.get_string()}
 
 
+class HackClubhouseAchievement(SingularEvent):
+    __tablename__ = 'hack_clubhouse_achievement'
+    __event_uuid__ = '62ce2e93-bfdc-4cae-af4c-54068abfaf02'
+    __payload_type__ = '(ss)'
+
+    achievement_id = Column(Unicode, nullable=False)
+    achievement_name = Column(Unicode, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {
+            'achievement_id': payload.get_child_value(0).get_string(),
+            'achievement_name': payload.get_child_value(1).get_string(),
+        }
+
+
+class HackClubhouseAchievementPoints(SingularEvent):
+    __tablename__ = 'hack_clubhouse_achievement_points'
+    __event_uuid__ = '86521913-bfa3-4d13-b511-a03d4e339d2f'
+    __payload_type__ = '(sii)'
+
+    skillset = Column(Unicode, nullable=False)
+    points = Column(Integer, nullable=False)
+    new_points = Column(Integer, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {
+            'skillset': payload.get_child_value(0).get_string(),
+            'points': payload.get_child_value(1).get_int32(),
+            'new_points': payload.get_child_value(2).get_int32(),
+        }
+
+
+class HackClubhouseChangePage(SingularEvent):
+    __tablename__ = 'hack_clubhouse_change_page'
+    __event_uuid__ = '2c765b36-a4c9-40ee-b313-dc73c4fa1f0d'
+    __payload_type__ = 's'
+
+    page = Column(Unicode, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {'page': payload.get_string()}
+
+
+class HackClubhouseEnterPathway(SingularEvent):
+    __tablename__ = 'hack_clubhouse_enter_pathway'
+    __event_uuid__ = '600c1cae-b391-4cb4-9930-ea284792fdfb'
+    __payload_type__ = 's'
+
+    pathway = Column(Unicode, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {'pathway': payload.get_string()}
+
+
+class HackClubhouseProgress(SingularEvent):
+    __tablename__ = 'hack_clubhouse_progress'
+    __event_uuid__ = '3a037364-9164-4b42-8c07-73bcc00902de'
+    __payload_type__ = 'a{sv}'
+
+    complete = Column(Boolean, nullable=False)
+    quest = Column(Unicode, nullable=False)
+    pathways = Column(ARRAY(Unicode, dimensions=1), nullable=False)
+    progress = Column(Numeric, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return get_asv_dict(payload)
```

Comment:
This will work as long as you don't remove any field from the payload, or change their types.

(I usually keep `get_asv_dict` for the cases where the columns aren't well defined, and we store the whole dict as a single `JSONB` field, because it is a bit fragile otherwise)

If you're ok with this caveat though, I see no reason to block this. :slightly_smiling_face:

Reply:
It's okay, the payload shouldn't change, it's not a dynamic payload, it'll be the same for all events.